java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -158,14 +158,14 @@ public static long serialize(WriteChannel out, Schema schema) throws IOException
    */
   public static long serialize(WriteChannel out, Schema schema, IpcOption option) throws IOException {
     long start = out.getCurrentPosition();
-    assert start % 8 == 0;
+    Preconditions.checkArgument(start % 8 == 0);
```

Comment:
more detail should be provided here (out is not aligned).  I'm 50/50 on whether this should be left as an assert since I think it is closer to programming bug, but seems ok to convert.

Reply:
same goes for the rest of the assert in this file.